src/main/java/org/folio/service/caches/InventoryCache.java

```diff
@@ -0,0 +1,115 @@
+package org.folio.service.caches;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
+import org.folio.rest.tools.utils.TenantTool;
+import org.folio.service.UserService;
+import org.folio.service.inventory.InventoryService;
+import org.springframework.stereotype.Component;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+
+@Component
+public class InventoryCache {
+  private static final Logger log = LogManager.getLogger();
+
+  private final AsyncCache<String, String> asyncCache;
+  private final AsyncCache<String, JsonObject> asyncJsonCache;
+  private static final String UNIQUE_CACHE_KEY_PATTERN = "%s_%s_%s";
+  private final InventoryService inventoryService;
+
+  public InventoryCache(InventoryService inventoryService) {
+    this.inventoryService = inventoryService;
+    asyncCache = Caffeine.newBuilder()
+      .expireAfterWrite(30, TimeUnit.SECONDS)
+      .executor(task -> Vertx.currentContext()
+        .runOnContext(v -> task.run()))
+      .buildAsync();
+
+    asyncJsonCache = Caffeine.newBuilder()
+      .expireAfterWrite(30, TimeUnit.SECONDS)
+      .executor(task -> Vertx.currentContext()
+        .runOnContext(v -> task.run()))
+      .buildAsync();
+  }
+
+  /**
+   * Retrieves {@link String} from cache by tenantId
+   * @param requestContext {@link RequestContext} connection params
+   * @return CompletableFuture with {@link String} ISBN UUID value
+   */
+  public Future<String> getProductTypeUuid(RequestContext requestContext) {
```

Comment:
Maybe we can rename method to add isbn to the name, because it can work only with isbn type name